packages/back-end/src/integrations/SqlIntegration.ts

```diff
@@ -3783,12 +3841,54 @@ export default abstract class SqlIntegration
     segment: SegmentInterface,
     baseIdType: string,
     idJoinMap: Record<string, string>,
+    factTableMap: FactTableMap,
     sqlVars?: SQLVars
   ) {
     // replace template variables
-    const segmentSql = sqlVars
-      ? compileSqlTemplate(segment.sql, sqlVars)
-      : segment.sql;
+    let segmentSql = "";
+
+    if (segment.sql) {
```

Comment:
Make type required instead of optional and add a migration to the segment model to set it if it's missing. Then, you can rely on that here.

The reason you want to use type instead of inferring is because someone could in theory, start with a SQL segment and then switch the type to FACT and the SQL field wouldn't get wiped out in Mongo. So it might be possible for SQL to be set, but it should still be treated like a fact table.

Yes, you'll still need additional empty checks for the SQL/fact table fields but that's ok.